test/fixtures/dialects/sparksql/numeric_literal.sql

```diff
@@ -28,3 +28,5 @@ WHERE
     AND baz > +3.E+3
     AND baz > -3.E-3D
     AND baz > -3.e-3d
+    AND baz > -+-1
+    AND baz > -+- 1
```

Comment:
I found a closely related issue involving multiple signs for column expressions. Worth fixing this too?

This fails to parse:
```
SELECT - - c1
FROM foo
```

Also this:
```
SELECT --c1
FROM foo
```

Reply:
I updated `SignedSegmentGrammar` instead to capture both (plus any future ones).

Note your last example is not valid as that (double negative with no whitespace) becomes a comment.